File: year2015/day02/python/solution.py
def main():
  print(part_1())
  print(part_2())


def part_1():
  return sum(2 * (l * w + l * h + w * h) + l * w for l, w, h in (sorted(box) for box in input))


def part_2():
  return sum(2 * (l + w) + l * w * h for l, w, h in (sorted(box) for box in input))


def setup(path):
  global input

  with open(path, 'r') as file:
    input = [[int(d) for d in row.split('x')] for row in file]


if __name__ == '__main__':
  setup('../puzzle.txt')
  main()

File: year2015/day02/java/Solution.java
import java.nio.file.*;
import java.util.*;

public class Solution {
    private static int[][] SIZES;

    public static void main(String[] args) { 
        try {
            setup(Files.readString(Paths.get("../puzzle.txt")));
            System.out.println(part1());
            System.out.println(part2());
        } catch(Exception e) {
            e.printStackTrace();
        }
    }
    
	public static int part1() {
        int sum = 0;
        for (int[] box : SIZES) {
            Arrays.sort(box);
            sum += 2 * (box[0] * box[1] + box[0] * box[2] + box[1] * box[2]) + box[0] * box[1];
        }
        return sum;
    }
    
    public static int part2() {
        int sum = 0;
        for (int[] box : SIZES) {
            Arrays.sort(box);
            sum += 2 * (box[0] + box[1]) + box[0] * box[1] * box[2];
        }
        return sum;
    }

    public static void setup(String input) {
        SIZES = Arrays.stream(input.split("\n")).map(line -> Arrays.stream(line.split("x")).mapToInt(Integer::parseInt).toArray()).toArray(int[][]::new);
    }
}